assets/js/dist/fade.js

```diff
@@ -1,40 +1,67 @@
-    el.addEventListener('click', function (e) {
-      e.preventDefault();
-      $('li').removeClass('current-btn');
-      $("#".concat(e.target.id)).addClass('current-btn');
-      $('#fade-wrapper').fadeOut(500, function () {
-        $('section').removeClass('display');
-        var sectionId = e.target.id.slice(4);
-        $("#".concat(sectionId)).addClass('display');
-
-        if ($(window).scrollTop() > mobileOffset) {
-          console.log('reset window');
-          window.scrollTo(0, mobileOffset);
-        }
-
-        $('#fade-wrapper').fadeIn(500, function () {
-          console.log("faded in");
-        });
-      });
+var headerElement = document.getElementById('header');
+var headerGeometry = headerElement.getBoundingClientRect();
+var stickyNavOffset = headerGeometry.height + window.scrollY + headerGeometry.y;
+var navElements = Array.from(document.getElementsByClassName('nav-btn'));
+var sectionElements = Array.from(document.getElementsByTagName('section'));
+var fadeElement = document.getElementById('fade-wrapper');
+
+function selectWithLocation(el, locationHash) {
+  return el.dataset.section === locationHash.substring(1);
+}
+
+function getActiveElements(locationHash) {
+  if (locationHash && locationHash !== '') {
+    var navElement = navElements.find(function (el) {
+      return selectWithLocation(el, locationHash);
     });
-  }); // Watch for scroll for fixed nav bar
-
-  $(window).scroll(function () {
-    if ($(window).scrollTop() > mobileOffset) {
-      $('nav').addClass('fixed-nav');
-      $('#wrapper').css('margin-top', '80px');
-    } else {
-      $('nav').removeClass('fixed-nav');
-      $('#wrapper').css('margin-top', '0');
-    }
+    var sectionElement = sectionElements.find(function (el) {
+      return selectWithLocation(el, locationHash);
+    });
+    return [navElement, sectionElement];
+  }
+
+  return [navElements[0], sectionElements[0]];
+} // Page load could be index.html, or index.html#contact etc.
+
+
+var activeElements = getActiveElements(window.location.hash);
+activeElements[0].classList.add('current-btn');
+activeElements[1].classList.add('display');
+window.addEventListener('hashchange', function (e) {
+  e.preventDefault();
+  navElements.forEach(function (el) {
+    el.classList.remove('current-btn');
   });
+  var elements = getActiveElements(window.location.hash);
+  elements[0].classList.add('current-btn');
+  fadeElement.classList.add('hidden');
+  var fadeDuration = getComputedStyle(fadeElement).transitionDuration;
+  var durationMs = parseFloat(fadeDuration.match(/\d+.\d+/)) * 1000;
+  window.setTimeout(function () {
+    sectionElements.forEach(function (el) {
+      el.classList.remove('display');
+    });
+    elements[1].classList.add('display');
+
+    if (window.scrollY > stickyNavOffset) {
+      window.scrollTo(0, stickyNavOffset);
+    }
+
+    fadeElement.classList.remove('hidden');
+    fadeElement.classList.add('visible');
+  }, durationMs);
```

Comment:
What happens if the user clicks a different nav button while we are waiting for the timeout? Does the animation look weird? I wonder if we need to stash the timeout ID somewhere and call`window.clearTimeout` if there is a timeout pending.

Reply:
I'll look into this ... I've seen it look a bit abrupt at times, but mostly it looks pretty good.

Reply:
Interestingly, you might think (as you did) that you'd get some strange behavior with the visibility/hidden classes getting added and removed on clicking another nav link mid timeout... or the previous click timing out in the middle of the new one etc.... 

But that doesn't seem to be happening... I don't quite understand it. Also it seems to display the correct new section even if logically it seems like the time hasn't yet elapsed for it to have made that determination...    

At any rate, I did a version with the `clearTimeout` which I think is more rigorous I'll show you next PR